File: src/main/java/com/study/platform/domain/chat/infrastructure/RedisChatSubscriber.java
package com.study.platform.domain.chat.infrastructure;

import com.study.platform.domain.chat.dto.response.ChatMessageResponse;
import com.study.platform.global.constant.WebSocketConstants;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.redis.connection.Message;
import org.springframework.data.redis.connection.MessageListener;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Component;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

@Slf4j
@Component
@RequiredArgsConstructor
public class RedisChatSubscriber implements MessageListener {

    private final SimpMessagingTemplate messagingTemplate;
    private final ObjectMapper objectMapper;

    @Override
    public void onMessage(Message message, byte[] pattern) {
        try {
            ChatMessageResponse response = objectMapper.readValue(message.getBody(), ChatMessageResponse.class);
            messagingTemplate.convertAndSend(
                    WebSocketConstants.CHAT_TOPIC_PREFIX + response.teamId(), response
            );
        } catch (IOException e) {
            log.error("Redis 채팅 메시지 처리 실패", e);
        }
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/chat/infrastructure/RedisChatSubscriber.kt
package com.study.platform.domain.chat.infrastructure

import com.study.platform.domain.chat.dto.response.ChatMessageResponse
import com.study.platform.global.constant.WebSocketConstants
import org.slf4j.LoggerFactory
import org.springframework.data.redis.connection.Message
import org.springframework.data.redis.connection.MessageListener
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Component
import tools.jackson.core.JacksonException
import tools.jackson.databind.ObjectMapper

@Component
class RedisChatSubscriber(
    private val messagingTemplate: SimpMessagingTemplate,
    private val objectMapper: ObjectMapper
) : MessageListener {

    private val log = LoggerFactory.getLogger(RedisChatSubscriber::class.java)!!

    override fun onMessage(message: Message, pattern: ByteArray?) {
        try {
            val response = objectMapper.readValue(message.body, ChatMessageResponse::class.java)
            messagingTemplate.convertAndSend(
                WebSocketConstants.CHAT_TOPIC_PREFIX + response.teamId, response
            )
        } catch (e: JacksonException) {
            log.error("Redis 채팅 메시지 처리 실패", e)
        }
    }
}
